diff -ru qgis-3.32.0-orig/CMakeLists.txt qgis-3.32.0/CMakeLists.txt
--- qgis-3.32.0-orig/CMakeLists.txt	2023-06-29 08:45:13.083130699 +1000
+++ qgis-3.32.0/CMakeLists.txt	2023-06-29 17:08:20.975609696 +1000
@@ -363,7 +363,18 @@
     message (SEND_ERROR "sqlite3 dependency was not found!")
   endif()
 
-  find_package(Protobuf REQUIRED)   # for decoding of vector tiles in MVT format
+  if(MSVC)
+    # for some reason on Windows we need to do this so abseil
+    # gets linked in (along with the protobuf::libprotobuf-lite change)
+    # but this breaks Unix.
+    # (see https://github.com/conda-forge/conda-forge-pinning-feedstock/issues/4075)
+    block()
+      set(CMAKE_FIND_PACKAGE_PREFER_CONFIG 1)
+      find_package(Protobuf REQUIRED)
+    endblock()
+  else()
+    find_package(Protobuf REQUIRED)   # for decoding of vector tiles in MVT format
+  endif()
   message(STATUS "Found Protobuf: ${Protobuf_LIBRARIES}")
   if (NOT Protobuf_PROTOC_EXECUTABLE)
     message (SEND_ERROR "Protobuf library's 'protoc' tool was not found!")
diff -ru qgis-3.32.0-orig/src/core/CMakeLists.txt qgis-3.32.0/src/core/CMakeLists.txt
--- qgis-3.32.0-orig/src/core/CMakeLists.txt	2023-06-29 08:45:13.555301061 +1000
+++ qgis-3.32.0/src/core/CMakeLists.txt	2023-06-29 08:55:37.311270163 +1000
@@ -2029,6 +2029,9 @@
   set_source_files_properties(${VECTOR_TILE_PROTO_SRCS} PROPERTIES COMPILE_FLAGS -w)
 endif()
 
+# now on all platforms (see https://github.com/conda-forge/conda-forge-pinning-feedstock/issues/4075)
+set_source_files_properties(${VECTOR_TILE_PROTO_SRCS} vectortile/qgsvectortilemvtdecoder.cpp vectortile/qgsvectortilemvtencoder.cpp vectortile/qgsvectortilewriter.cpp PROPERTIES COMPILE_DEFINITIONS PROTOBUF_USE_DLLS)
+
 # install headers
 # install qgsconfig.h and plugin.h here so they can get into
 # the OS X framework target
@@ -2375,7 +2378,7 @@
   ${EXPAT_LIBRARY}
   ${SQLITE3_LIBRARY}
   ${LIBZIP_LIBRARY}
-  ${Protobuf_LITE_LIBRARY}
+  protobuf::libprotobuf-lite
   ${ZLIB_LIBRARIES}
   ${EXIV2_LIBRARY}
 )
