diff -ru qgis-3.32.0-orig/CMakeLists.txt qgis-3.32.0/CMakeLists.txt
--- qgis-3.32.0-orig/CMakeLists.txt	2023-06-28 10:07:16.955017635 +1000
+++ qgis-3.32.0/CMakeLists.txt	2023-06-28 10:44:45.367369908 +1000
@@ -363,6 +363,8 @@
     message (SEND_ERROR "sqlite3 dependency was not found!")
   endif()
 
+  # see https://github.com/conda-forge/conda-forge-pinning-feedstock/issues/4075
+  set(CMAKE_FIND_PACKAGE_PREFER_CONFIG 1)
   find_package(Protobuf REQUIRED)   # for decoding of vector tiles in MVT format
   message(STATUS "Found Protobuf: ${Protobuf_LIBRARIES}")
   if (NOT Protobuf_PROTOC_EXECUTABLE)
@@ -1160,3 +1162,11 @@
 if (UNIX AND NOT APPLE)
   add_subdirectory(linux)
 endif()
+
+# see https://github.com/conda-forge/conda-forge-pinning-feedstock/issues/4075
+if(Protobuf_PROTOC_EXECUTABLE)
+  set_target_properties(protobuf::protoc PROPERTIES
+  IMPORTED_LOCATION_RELEASE "${Protobuf_PROTOC_EXECUTABLE}"
+  )
+endif()
+
diff -ru qgis-3.32.0-orig/src/core/CMakeLists.txt qgis-3.32.0/src/core/CMakeLists.txt
--- qgis-3.32.0-orig/src/core/CMakeLists.txt	2023-06-28 10:07:17.434978947 +1000
+++ qgis-3.32.0/src/core/CMakeLists.txt	2023-06-28 10:07:28.562079765 +1000
@@ -2029,6 +2029,9 @@
   set_source_files_properties(${VECTOR_TILE_PROTO_SRCS} PROPERTIES COMPILE_FLAGS -w)
 endif()
 
+# now on all platforms (see https://github.com/conda-forge/conda-forge-pinning-feedstock/issues/4075)
+set_source_files_properties(${VECTOR_TILE_PROTO_SRCS} vectortile/qgsvectortilemvtdecoder.cpp vectortile/qgsvectortilemvtencoder.cpp vectortile/qgsvectortilewriter.cpp PROPERTIES COMPILE_DEFINITIONS PROTOBUF_USE_DLLS)
+
 # install headers
 # install qgsconfig.h and plugin.h here so they can get into
 # the OS X framework target
