From f5397398a0d697bcefb10d7be589fc215f3abc6a Mon Sep 17 00:00:00 2001
From: Ista Zahn <izahn@hbs.edu>
Date: Mon, 19 Sep 2022 06:32:25 -0400
Subject: [PATCH] use protobuf dlls

---
 src/core/CMakeLists.txt | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/core/CMakeLists.txt b/src/core/CMakeLists.txt
index 7537413..8eed55d 100644
--- a/src/core/CMakeLists.txt
+++ b/src/core/CMakeLists.txt
@@ -1813,6 +1813,10 @@ if (NOT MSVC)
   set_source_files_properties(${VECTOR_TILE_PROTO_SRCS} PROPERTIES COMPILE_FLAGS -w)
 endif()
 
+if (MSVC)
+  set_source_files_properties(${VECTOR_TILE_PROTO_SRCS} PROTOBUF_USE_DLLS)
+endif()
+
 # install headers
 # install qgsconfig.h and plugin.h here so they can get into
 # the OS X framework target
-- 
2.37.3

