--- CMakeLists.txt.orig	2024-06-24 07:51:08.157059805 +1000
+++ CMakeLists.txt	2024-06-24 07:52:51.496436739 +1000
@@ -374,12 +374,17 @@
   endif()
 
   find_package(Protobuf CONFIG)
-  find_package(Protobuf REQUIRED)
 
   message(STATUS "Found Protobuf: ${Protobuf_LIBRARIES}")
   if (NOT Protobuf_PROTOC_EXECUTABLE)
     message (SEND_ERROR "Protobuf library's 'protoc' tool was not found!")
   endif()
+
+  # for cross compilation  
+  set_target_properties(protobuf::protoc PROPERTIES
+      IMPORTED_LOCATION_RELEASE "${Protobuf_PROTOC_EXECUTABLE}"
+  )
+
   find_package(ZLIB REQUIRED)       # for decompression of vector tiles in MBTiles file
   message(STATUS "Found zlib: ${ZLIB_LIBRARIES}")
 
