--- python/__init__.py.orig	2022-09-21 08:20:39.694757195 +1000
+++ python/__init__.py	2022-09-21 08:21:02.862041298 +1000
@@ -72,7 +72,7 @@
 
     if sys.version_info[0] > 3 or (sys.version_info[0] == 3 and sys.version_info[1] >= 9):
         for p in os.getenv("PATH").split(";"):
-            if os.path.exists(p):
+            if os.path.exists(p) and p != '.':
                 os.add_dll_directory(p)
 
 from qgis.PyQt import QtCore
