--- python/__init__.py.orig	2022-09-21 08:20:39.694757195 +1000
+++ python/__init__.py	2022-09-21 08:59:11.269618728 +1000
@@ -73,7 +73,7 @@
     if sys.version_info[0] > 3 or (sys.version_info[0] == 3 and sys.version_info[1] >= 9):
         for p in os.getenv("PATH").split(";"):
             if os.path.exists(p):
-                os.add_dll_directory(p)
+                os.add_dll_directory(os.path.abspath(p))
 
 from qgis.PyQt import QtCore
 
