--- python/CMakeLists.txt.orig	2023-07-05 12:59:35.366379903 +1000
+++ python/CMakeLists.txt	2023-07-05 12:59:54.574852617 +1000
@@ -176,12 +176,6 @@
 BUILD_SIP_PYTHON_MODULE(qgis._core core/core.sip ${cpp_files} "" qgis_core)
 set(SIP_CORE_CPP_FILES ${cpp_files})
 
-if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
-  # Bad hack to fix compilation with gcc 11 - for some reason it's ignoring
-  # the c++ standard version set for the target in BUILD_SIP_PYTHON_MODULE!
-  add_definitions(-std=c++14)
-endif()
-
 if((UNIX OR MINGW) AND SIP_VERSION_STR VERSION_GREATER_EQUAL 4.10 AND SIP_VERSION_STR VERSION_LESS 5.0)
   set(SIP_EXTRA_OPTIONS -P ${SIP_EXTRA_OPTIONS})
   add_definitions(-Dprotected=public)
