--- CMakeLists.txt.orig	2023-11-13 08:12:57.054099434 +1000
+++ CMakeLists.txt	2023-11-13 08:32:27.445388907 +1000
@@ -366,11 +366,29 @@
     message (SEND_ERROR "sqlite3 dependency was not found!")
   endif()
 
-  find_package(Protobuf REQUIRED)   # for decoding of vector tiles in MVT format
+  # First look for protobuf-config.cmake, and only later use
+  # FindProtobuf.cmake shipped with CMake, for compatibility
+  # with protobuf >= 22
+  set(protobuf_MODULE_COMPATIBLE TRUE CACHE INTERNAL "")
+  find_package(Protobuf CONFIG)   # for decoding of vector tiles in MVT format
+  if(Protobuf_FOUND)
+     message(STATUS "Found protobuf via cmake config")
+     if(Protobuf_PROTOC_EXECUTABLE)
+      set_target_properties(protobuf::protoc PROPERTIES
+        IMPORTED_LOCATION_RELEASE "${Protobuf_PROTOC_EXECUTABLE}"
+      )
+    endif()
+ else()
+     message(WARNING "Falling back to cmake FindProtobuf as Protobuf was not found via CONFIG")
+     find_package(Protobuf REQUIRED)
+ endif()  
   message(STATUS "Found Protobuf: ${Protobuf_LIBRARIES}")
   if (NOT Protobuf_PROTOC_EXECUTABLE)
     message (SEND_ERROR "Protobuf library's 'protoc' tool was not found!")
   endif()
+  if(Protobuf_VERSION GREATER_EQUAL 4.23)
+    set(Protobuf_LITE_LIBRARY protobuf::libprotobuf-lite)
+  endif()
   find_package(ZLIB REQUIRED)       # for decompression of vector tiles in MBTiles file
   message(STATUS "Found zlib: ${ZLIB_LIBRARIES}")
 
