diff --git a/src/providers/spatialite/CMakeLists.txt b/src/providers/spatialite/CMakeLists.txt
index d91b6893..3afe77c3 100644
--- a/src/providers/spatialite/CMakeLists.txt
+++ b/src/providers/spatialite/CMakeLists.txt
@@ -63,6 +63,7 @@ ADD_LIBRARY (spatialiteprovider MODULE ${SPATIALITE_SRCS} ${SPATIALITE_MOC_SRCS}
 
 TARGET_LINK_LIBRARIES(spatialiteprovider
   qgis_core
+  ${SQLITE3_LIBRARY}
   ${SPATIALITE_LIBRARY}
 )
 
