--- CMakeLists.txt.orig	2022-12-11 11:39:12.069267296 +1000
+++ CMakeLists.txt	2022-12-11 11:39:52.963609482 +1000
@@ -989,7 +989,8 @@
   include(SIPMacros)
 
   set(SIP_INCLUDES ${PYQT_SIP_DIR} ${CMAKE_SOURCE_DIR}/python)
-  set(SIP_CONCAT_PARTS 20)
+  set(SIP_CONCAT_PARTS 48)
+  set(SIP_BUILD_EXTRA_OPTIONS "--verbose")
 
   if (NOT BINDINGS_GLOBAL_INSTALL)
     set(Python_SITEARCH ${QGIS_DATA_DIR}/python)
