--- src/providers/pdal/CMakeLists.txt.orig	2021-03-11 13:11:08.017811000 +1000
+++ src/providers/pdal/CMakeLists.txt	2021-03-11 13:11:50.443733600 +1000
@@ -105,6 +105,7 @@
 set_target_properties(untwine PROPERTIES
   RUNTIME_OUTPUT_DIRECTORY ${QGIS_OUTPUT_DIRECTORY}/${QGIS_LIBEXEC_SUBDIR}
 )
+target_compile_definitions(untwine PRIVATE WIN32_LEAN_AND_MEAN)
 
 target_link_libraries (untwine
   ${PDAL_LIBRARIES}
