diff -u -r qgis-3.18.2-orig/src/app/CMakeLists.txt qgis-3.18.2/src/app/CMakeLists.txt
--- qgis-3.18.2-orig/src/app/CMakeLists.txt	2021-04-19 11:26:26.355427047 +1000
+++ qgis-3.18.2/src/app/CMakeLists.txt	2021-04-19 11:27:48.408609210 +1000
@@ -501,7 +501,7 @@
   libdxfrw
 )
 
-target_compile_definitions(qgis_app PRIVATE "-DQT_NO_FOREACH")
+target_compile_definitions(qgis_app PRIVATE "-DQT_NO_FOREACH -DWIN32_LEAN_AND_MEAN")
 
 if (WITH_BINDINGS)
   target_link_libraries(qgis_app qgispython)
diff -u -r qgis-3.18.2-orig/src/app/qgisapp.h qgis-3.18.2/src/app/qgisapp.h
--- qgis-3.18.2-orig/src/app/qgisapp.h	2021-04-19 11:26:26.363427161 +1000
+++ qgis-3.18.2/src/app/qgisapp.h	2021-04-19 11:28:55.353573619 +1000
@@ -183,6 +183,8 @@
 
 #ifdef Q_OS_WIN
 #include <windows.h>
+// because of WIN32_LEAN_AND_MEAN we need to include this explicitly
+#include <shellapi.h>
 #endif
 
 class QgsLegendFilterButton;
diff -u -r qgis-3.18.2-orig/src/providers/pdal/CMakeLists.txt qgis-3.18.2/src/providers/pdal/CMakeLists.txt
--- qgis-3.18.2-orig/src/providers/pdal/CMakeLists.txt	2021-04-19 11:26:26.523429467 +1000
+++ qgis-3.18.2/src/providers/pdal/CMakeLists.txt	2021-04-19 11:26:59.655906800 +1000
@@ -105,6 +105,7 @@
 set_target_properties(untwine PROPERTIES
   RUNTIME_OUTPUT_DIRECTORY ${QGIS_OUTPUT_DIRECTORY}/${QGIS_LIBEXEC_SUBDIR}
 )
+target_compile_definitions(untwine PRIVATE WIN32_LEAN_AND_MEAN)
 
 target_link_libraries (untwine
   ${PDAL_LIBRARIES}
