--- "cmake\\FindSpatialindex.cmake.orig"	2018-03-23 15:26:53.732177300 +1000
+++ "cmake\\FindSpatialindex.cmake"	2018-03-23 15:31:00.115200000 +1000
@@ -20,7 +20,7 @@
   PATH_SUFFIXES spatialindex
   )
 
-FIND_LIBRARY(SPATIALINDEX_LIBRARY NAMES spatialindex_i spatialindex PATHS
+FIND_LIBRARY(SPATIALINDEX_LIBRARY NAMES spatialindex_i spatialindex "spatialindex-$ENV{ARCH}" PATHS
   /usr/lib
   /usr/local/lib
   "$ENV{LIB_DIR}/lib"
