--- ./src/app/qgisapp.h.orig	2021-04-15 09:43:22.449631566 +1000
+++ ./src/app/qgisapp.h	2021-04-15 09:45:03.603113958 +1000
@@ -182,6 +182,9 @@
 #include <QTapAndHoldGesture>
 
 #ifdef Q_OS_WIN
+// ensure winsock2.h is included first to prevent winsock.h being
+// included by Windows.h (prevents conflicts with pdal etc which requres winsock2.h)
+#include <winsock2.h>
 #include <windows.h>
 #endif
 
