--- CMakeLists.txt.orig	2020-12-08 11:35:57.599824865 +1000
+++ CMakeLists.txt	2020-12-08 11:36:27.512232326 +1000
@@ -1004,7 +1004,7 @@
   if(NOT DEFINED CMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_NO_WARNINGS)
     set(CMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_NO_WARNINGS ON)
   endif()
-  include(InstallRequiredSystemLibraries)
+  #include(InstallRequiredSystemLibraries)
 
   set(CPACK_PACKAGE_DESCRIPTION_SUMMARY "QGIS")
   set(CPACK_PACKAGE_VENDOR "Open Source Geospatial Foundation")
