diff -r -u qgis-3.16.1-orig/CMakeLists.txt qgis-3.16.1/CMakeLists.txt
--- qgis-3.16.1-orig/CMakeLists.txt	2020-12-19 10:20:52.080327799 +1000
+++ qgis-3.16.1/CMakeLists.txt	2020-12-19 11:05:45.881051964 +1000
@@ -677,8 +677,12 @@
       set (DEFAULT_CGIBIN_SUBDIR .)
     endif()
   else()
+  
+    if(APPLE)
+        set(QGIS_MACAPP_FRAMEWORK TRUE CACHE BOOL "Build as a framework on OSX")
+    endif()
 
-    if (APPLE)
+    if (APPLE AND QGIS_MACAPP_FRAMEWORK)
       if (POLICY CMP0042) # in CMake 3.0.0+
         set (CMAKE_MACOSX_RPATH OFF) # otherwise ON by default
       endif()
@@ -751,6 +755,14 @@
       set (DEFAULT_QML_SUBDIR     qml)
 
       set (DEFAULT_SERVER_MODULE_SUBDIR ${DEFAULT_LIBEXEC_SUBDIR}/server)
+      
+      # QGIS_MACAPP_FRAMEWORK=FALSE
+      if(APPLE)
+        set (QGIS_MACAPP_BUNDLE -1)
+        set (CMAKE_FRAMEWORK FALSE)
+        set (QGIS_INSTALL_PREFIX ${CMAKE_INSTALL_PREFIX})
+      endif()
+ 
     endif()
 
   endif()
diff -r -u qgis-3.16.1-orig/src/3d/CMakeLists.txt qgis-3.16.1/src/3d/CMakeLists.txt
--- qgis-3.16.1-orig/src/3d/CMakeLists.txt	2020-12-19 10:20:52.640335998 +1000
+++ qgis-3.16.1/src/3d/CMakeLists.txt	2020-12-19 10:29:07.623522527 +1000
@@ -222,7 +222,7 @@
 
 set(QGIS_3D_HDRS ${QGIS_3D_HDRS} ${CMAKE_CURRENT_BINARY_DIR}/qgis_3d.h)
 
-if(NOT APPLE)
+if(NOT APPLE OR NOT QGIS_MACAPP_FRAMEWORK)
   install(FILES ${QGIS_3D_HDRS} DESTINATION ${QGIS_INCLUDE_DIR})
 else()
   set_target_properties(qgis_3d PROPERTIES
diff -r -u qgis-3.16.1-orig/src/analysis/CMakeLists.txt qgis-3.16.1/src/analysis/CMakeLists.txt
--- qgis-3.16.1-orig/src/analysis/CMakeLists.txt	2020-12-19 10:20:52.648336115 +1000
+++ qgis-3.16.1/src/analysis/CMakeLists.txt	2020-12-19 10:31:03.881197981 +1000
@@ -465,7 +465,7 @@
 
 set(QGIS_ANALYSIS_HDRS ${QGIS_ANALYSIS_HDRS} ${CMAKE_CURRENT_BINARY_DIR}/qgis_analysis.h)
 
-if(NOT APPLE)
+if(NOT APPLE OR NOT QGIS_MACAPP_FRAMEWORK)
   install(FILES ${QGIS_ANALYSIS_HDRS} DESTINATION ${QGIS_INCLUDE_DIR})
 else()
   set_target_properties(qgis_analysis PROPERTIES
diff -r -u qgis-3.16.1-orig/src/core/CMakeLists.txt qgis-3.16.1/src/core/CMakeLists.txt
--- qgis-3.16.1-orig/src/core/CMakeLists.txt	2020-12-19 10:20:52.684336642 +1000
+++ qgis-3.16.1/src/core/CMakeLists.txt	2020-12-19 10:31:35.305650339 +1000
@@ -1654,7 +1654,7 @@
 endif()
 
 
-if(NOT APPLE)
+if(NOT APPLE OR NOT QGIS_MACAPP_FRAMEWORK)
   install(FILES ${QGIS_CORE_HDRS} DESTINATION ${QGIS_INCLUDE_DIR})
 else()
 
diff -r -u qgis-3.16.1-orig/src/gui/CMakeLists.txt qgis-3.16.1/src/gui/CMakeLists.txt
--- qgis-3.16.1-orig/src/gui/CMakeLists.txt	2020-12-19 10:20:52.736337405 +1000
+++ qgis-3.16.1/src/gui/CMakeLists.txt	2020-12-19 10:33:27.995270986 +1000
@@ -1383,11 +1383,13 @@
 
 set(QGIS_GUI_HDRS ${QGIS_GUI_HDRS} ${CMAKE_CURRENT_BINARY_DIR}/qgis_gui.h)
 
-if(NOT APPLE)
+if(NOT APPLE OR NOT QGIS_MACAPP_FRAMEWORK)
   if (WIN32 )
     include_directories(${CMAKE_SOURCE_DIR}/src/native/win)
   endif()
-  if (UNIX)
+  elseif (APPLE)
+    include_directories(${CMAKE_SOURCE_DIR}/src/native/mac)
+  elseif (UNIX)
     include_directories(${CMAKE_SOURCE_DIR}/src/native/linux)
   endif()
   install(FILES ${QGIS_GUI_HDRS} DESTINATION ${QGIS_INCLUDE_DIR})
@@ -1474,7 +1476,7 @@
   PUBLIC_HEADER DESTINATION ${QGIS_INCLUDE_DIR})
 
 # ui headers to install, don't exist at target definition time, install manually
-if (APPLE)
+if (APPLE AND QGIS_MACAPP_FRAMEWORK)
   install(FILES ${QGIS_GUI_UI_HDRS} DESTINATION ${QGIS_FW_SUBDIR}/qgis_gui.framework/Headers)
 else()
   install(FILES ${QGIS_GUI_UI_HDRS} DESTINATION ${QGIS_INCLUDE_DIR})
diff -r -u qgis-3.16.1-orig/src/native/CMakeLists.txt qgis-3.16.1/src/native/CMakeLists.txt
--- qgis-3.16.1-orig/src/native/CMakeLists.txt	2020-12-19 10:20:52.772337932 +1000
+++ qgis-3.16.1/src/native/CMakeLists.txt	2020-12-19 10:33:52.443622308 +1000
@@ -107,7 +107,7 @@
 
 set(QGIS_NATIVE_HDRS ${QGIS_NATIVE_HDRS} ${CMAKE_CURRENT_BINARY_DIR}/qgis_native.h)
 
-if(NOT APPLE)
+if(NOT APPLE OR NOT QGIS_MACAPP_FRAMEWORK)
   install(FILES ${QGIS_NATIVE_HDRS} DESTINATION ${QGIS_INCLUDE_DIR})
 else()
   set_target_properties(qgis_native PROPERTIES
diff -r -u qgis-3.16.1-orig/src/providers/grass/CMakeLists.txt qgis-3.16.1/src/providers/grass/CMakeLists.txt
--- qgis-3.16.1-orig/src/providers/grass/CMakeLists.txt	2020-12-19 10:20:52.812338517 +1000
+++ qgis-3.16.1/src/providers/grass/CMakeLists.txt	2020-12-19 10:34:36.084249198 +1000
@@ -93,7 +93,7 @@
 
     set_target_properties(qgisgrass${GRASS_BUILD_VERSION} PROPERTIES
       CLEAN_DIRECT_OUTPUT 1
-      FRAMEWORK 1
+      FRAMEWORK QGIS_MACAPP_FRAMEWORK
       FRAMEWORK_VERSION "${CPACK_PACKAGE_VERSION_MAJOR}.${CPACK_PACKAGE_VERSION_MINOR}"
       MACOSX_FRAMEWORK_INFO_PLIST "${CMAKE_SOURCE_DIR}/mac/framework.info.plist.in"
       MACOSX_FRAMEWORK_SHORT_VERSION_STRING ${COMPLETE_VERSION}
diff -r -u qgis-3.16.1-orig/src/quickgui/CMakeLists.txt qgis-3.16.1/src/quickgui/CMakeLists.txt
--- qgis-3.16.1-orig/src/quickgui/CMakeLists.txt	2020-12-19 10:20:52.824338694 +1000
+++ qgis-3.16.1/src/quickgui/CMakeLists.txt	2020-12-19 10:35:13.260783023 +1000
@@ -120,7 +120,7 @@
 )
 
 # Installation
-if(NOT APPLE)
+if(NOT APPLE OR NOT QGIS_MACAPP_FRAMEWORK)
   install(FILES ${CMAKE_CURRENT_BINARY_DIR}/qgis_quick.h ${QGIS_QUICK_GUI_HDRS} ${QGIS_QUICK_GUI_MOC_HDRS} DESTINATION ${QGIS_INCLUDE_DIR})
 else()
   set_target_properties(qgis_quick PROPERTIES
