diff --git debian/python3-qgis-common.install debian/python3-qgis-common.install
index 0c0bb9767f..320f178d59 100644
--- debian/python3-qgis-common.install
+++ debian/python3-qgis-common.install
@@ -1,5 +1,5 @@
 usr/lib/python*/*-packages/PyQt?/uic/widget-plugins/qgis_customwidgets.py
-usr/lib/python*/*-packages/qgis/*.py
+usr/lib/python*/*-packages/qgis/*.py*
 usr/lib/python*/*-packages/qgis/PyQt/*
 usr/lib/python*/*-packages/qgis/processing/*
 usr/lib/python*/*-packages/qgis/testing/*
diff --git debian/python3-qgis.install debian/python3-qgis.install
index 2425a3080d..24e4813935 100644
--- debian/python3-qgis.install
+++ debian/python3-qgis.install
@@ -1,10 +1,6 @@
 usr/lib/python*/*-packages/qgis/3d/__init__.py
-usr/lib/python*/*-packages/qgis/_3d.so
-usr/lib/python*/*-packages/qgis/_analysis.so
-usr/lib/python*/*-packages/qgis/_core.so
-usr/lib/python*/*-packages/qgis/_gui.so
-usr/lib/python*/*-packages/qgis/_server.so
 usr/lib/python*/*-packages/qgis/analysis/__init__.py
 usr/lib/python*/*-packages/qgis/core/__init__.py
 usr/lib/python*/*-packages/qgis/gui/__init__.py
 usr/lib/python*/*-packages/qgis/server/__init__.py
+usr/lib/python*/*-packages/qgis/*.so
diff --git ms-windows/osgeo4w/package.cmd ms-windows/osgeo4w/package.cmd
index 32f2c35617..cf34ed6f1e 100644
--- ms-windows/osgeo4w/package.cmd
+++ ms-windows/osgeo4w/package.cmd
@@ -332,6 +332,7 @@ if errorlevel 1 (echo tar common failed & goto error)
 	"apps/%PACKAGENAME%/resources/server/" ^
 	"apps/%PACKAGENAME%/server/" ^
 	"apps/%PACKAGENAME%/python/qgis/_server.pyd" ^
+	"apps/%PACKAGENAME%/python/qgis/_server.pyi" ^
 	"apps/%PACKAGENAME%/python/qgis/server/" ^
 	"httpd.d/httpd_%PACKAGENAME%.conf.tmpl" ^
 	"etc/postinstall/%PACKAGENAME%-server.bat" ^
@@ -367,6 +368,7 @@ if not exist %ARCH%\release\qgis\%PACKAGENAME% mkdir %ARCH%\release\qgis\%PACKAG
 	--exclude-from exclude ^
 	--exclude "*.pyc" ^
 	--exclude "apps/%PACKAGENAME%/python/qgis/_server.pyd" ^
+	--exclude "apps/%PACKAGENAME%/python/qgis/_server.pyi" ^
 	--exclude "apps/%PACKAGENAME%/python/qgis/_server.lib" ^
 	--exclude "apps/%PACKAGENAME%/python/qgis/server" ^
 	--exclude "apps/%PACKAGENAME%/server/" ^
diff --git python/CMakeLists.txt python/CMakeLists.txt
index 2e081dc20b..5f1f468d55 100644
--- python/CMakeLists.txt
+++ python/CMakeLists.txt
@@ -440,3 +440,8 @@ foreach(module ${PY_MODULES})
     install(DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/${module}/auto_generated DESTINATION ${SIP_DEFAULT_SIP_DIR}/qgis/${module})
   endif()
 endforeach(module)
+
+if((${SIP_VERSION_STR} VERSION_EQUAL 4.18) OR (${SIP_VERSION_STR} VERSION_GREATER 4.18))
+  file(GLOB PY_PYIS ${QGIS_PYTHON_OUTPUT_DIRECTORY}/*.pyi)
+  install(FILES ${PY_PYIS} DESTINATION ${QGIS_PYTHON_DIR})
+endif()
